proposed/activity-improvements.md

```diff
@@ -0,0 +1,465 @@
+    {
+        // optionally the activity can be enriched with extra data
+        activity?.AddTag("userId", userId);
+        activity?.AddTag("fooParam", foo);
+
+        // the developer's original code goes here
+    }
+```
+
+### 2. Monitoring agent wants to listen to activities created from specific ActivitySource
+
+A monitoring agent can create a listener which represents their subscription to activity callbacks like this.
+
+```C#
+ActivityListener listener = new ActivityListener
+{
+    ActivityStopped = (activity) => { ... },
+    ActivityStarted = (activity) => { ... },
+    ShouldListenTo = (activitySource) => ...,
+    GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> options) => ...,
+    GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> options) => ... 
+};
+ActivitySource.AddActivityListener(listener);
+
+...
+    
+listener.Dispose();  
+```
+
+It is up to the listener to do extract information from the Activity object and respond to the callbacks however they like, probably by serializing a message to a remote service that will aggregate the telemetry.
+
+[TODO]: We could flesh out what we expect it look like end-to-end to use this with OpenTelemetry and perhaps some UI? It would help newcomers understand but I don't think it is a blocking issue for the current set of collaborators.
+
+Requirements:
+=============
+
+### Goals
+
+#### 1. All Activities can be collected
+
+The current design of DiagnosticListener and the conventions used by existing instrumented libraries expect the collection agent to have significant knowledge about each instrumented library. We need to make it possible for a general purpose collection agent to collect distributed trace telemetry despite having none of this advance knowledge. This also implies that any developer can add new Activity instrumentation to their app or library and the agent is capable or recording it automatically.
+
+#### 2. Activity collection can be filtered with a predicate on name
+
+We are assuming that most Activity information is desirable by default, but the monitoring agent should be able to filter out noisy or problematic instrumentation. The agent can efficiently decide if it wants to listen to Activities which created from specific source/Component (ActivitySource).
+
+#### 3. Activity needs API representations of OpenTelemetry Span concepts
+
+What .Net calls Activity, OpenTelemetry calls Span. The OT span can express a broader set of information than Activity currently can. We want to create parity so that developers who want to emit certain OpenTelemetry data points can still use Activity APIs to accomplish that.
+
+#### 4. API usability
+
+Some of the existing patterns of API usage are awkward and don't appeal to many developers. We'd like to simplify usage wherever possible so that the APIs feel more approachable to a broader set of .Net developers.
```

Comment:
```suggestion
Some of the existing patterns of API usage are awkward and don't appeal to many developers. We'd like to simplify usage wherever possible so that the APIs feel more approachable to a broader set of .NET developers.
```